### ISSUE TITLE
Fix  test "testEnvironment reset for next test"

### ISSUE THREAD

**Issue by Krinkle** [qunitjs/qunit#477]
There's a problem with test "testEnvironment reset for next test".

It passes when run on a clean sessionStorage, but if one of the tests failed earlier, this particular test will now fail because we change the execution order.

This test relies on being run after the previous one in the test suite.

I tried hacking around it with callbacks (e.g. deferred) but it doesn't work because when the 2nd one runs first, as it will wait forever (we change order, but we don't run in parallel, can't yet because of global state).

Looks like something we need to rethink, perhaps solvable by allowing them to run in parallel after we refactor Assertion.
